core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -478,12 +500,21 @@ private[spark] class ExecutorAllocationManager(
     /**
      * An estimate of the total number of pending tasks remaining for currently running stages. Does
      * not account for tasks which may have failed and been resubmitted.
+     *
+     * Note: This is not thread-safe without the caller owning the `allocationManager` lock.
      */
     def totalPendingTasks(): Int = {
       stageIdToNumTasks.map { case (stageId, numTasks) =>
         numTasks - stageIdToTaskIndices.get(stageId).map(_.size).getOrElse(0)
       }.sum
     }
+
+    /**
+     * Note: This is not thread-safe without the caller owning the `allocationManager` lock.
+     */
```

Comment:
This isn't what the function does. Even though it's obvious we should keep the java docs descriptive
```
Return true if an executor is not currently running a task, and false otherwise.
Note: ...
```